docs/_docs/configuration.md

```diff
@@ -461,7 +461,11 @@ The default value for `JEKYLL_ENV` is `development`. Therefore if you omit `JEKY
 
 Your environment values can be anything you want (not just `development` or `production`). Some elements you might want to hide in development environments include Disqus comment forms or Google Analytics. Conversely, you might want to expose an "Edit me in GitHub" button in a development environment but not include it in production environments.
 
-By specifying the option in the build command, you avoid having to change values in your configuration files when moving from one environment to another.
+<div class="note info">
+  <p>
+    <code>JEKYLL_ENV</code> allows for environment specific content. In case you need environment specific configuration options, use the <a href="#build-command-options">build command option</a>, for example <code>--config _config.yml,_config.development.yml</code>. Settings in later files override settings in earlier files.
```

Comment:
*Novice users* are probably going to be confused as to what qualifies as `environment specific content` and `environment specific configuration options`. You may want to add a paragraph explaining those with a simple example.

Existing templates *may have* a Liquid block like the following:
```
{% if jekyll.environment == "production" %}
  Render this!
{% else %}
  Render something else
{% endif %}
```
The above is not a wrong usage, and `jekyll.environment` can only be altered via `JEKYLL_ENV` env.

Reply:
@ashmaroli My goal is to stitch the missing link between the "build-command-options" and this area. The section above your comment explains in detail how jekyll.environment works; I did not change this. My change is at the end of the section, linking to a different way to solve very similar things.

You comment feels like a rework of the whole paragraph https://github.com/jekyll/jekyll/blob/67c597c5b42018843e70fe514cfe6c8104526ac0/docs/_docs/configuration.md#specifying-a-jekyll-environment-at-build-time which IMO should be a different PR.